woocommerce/woocommerce-android
# [Screenshot automation] promo screenshots

Part of #6833


⚠️ Please don't merge until #6834 is merged.

### Description
This PR adds the following changes:
1. Adds a lane for downloading translated promo strings from GlotPress.
2. Adds a lane to generate promo screenshots.

The work is based on what we have in WPAndroid and WCiOS with some simplifications/changes, please check my comments on the code for questions/clarifications.

Question: do you think we should move the screenshots lanes to a separate file like WPAndroid?

### Testing instructions
The screenshot generation is still broken for some locales, so we'll work with just a limited number of locales for this test, when we get to the third subtask of this task, I'll try to update the tests and fix the broken ones.

1. Install the required dependencies with the following commands:
     - `brew install imagemagick automattic/build-tools/drawText`
     - `bundle install --with screenshots`
     - Install the font Proxima Nova from [here](https://drive.google.com/drive/folders/18vuAG0LWl1nPFTBBO4Ghhs3YTG1HHG_n).
2. Download promo screenshots metadata:
`bundle exec fastlane download_promo_strings`
3. Generate screenshots by running the following command:
`bundle exec fastlane take_screenshots locales:en-US,fr-FR`
4. Generate promo screenshots by running the following command:
`bundle exec fastlane create_promo_screenshots`

The last command should generate promo screenshots in the directory: `screenshots/promo_screenshots` for all the languages we support (since we generated raw screenshots just for 2 locales for the test, we will have just these two locales for the promo screenshots as well).

### Images/gif
| | | | |
| - | - | - | - |
| ![Pixel 3-01](https://user-images.githubusercontent.com/1657201/176947839-c2a4e260-e7e8-46d8-ba66-8c136f427297.png) | ![Pixel 3-01_b](https://user-images.githubusercontent.com/1657201/176947855-c85a11b9-eef0-4558-b7c4-30d857b66ecc.png) | ![Pixel 3-01](https://user-images.githubusercontent.com/1657201/176947861-17189be6-f645-4913-9dac-1a3a771cd3a3.png) | ![Pixel 3-01_b](https://user-images.githubusercontent.com/1657201/176947882-ee65e59f-15ee-4e27-adb3-0794990ceff8.png) |
